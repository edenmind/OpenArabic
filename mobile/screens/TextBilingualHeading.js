import { Caption, Subheading, Title } from 'react-native-paper'
import React, { Fragment } from 'react'

import PropTypes from 'prop-types'
/* eslint-disable import/namespace */
import { StyleSheet } from 'react-native'

const style = StyleSheet.create({
  author: {
    textAlign: 'center'
  },
  source: {
    paddingBottom: 10,
    textAlign: 'center'
  },
  title: {
    paddingTop: 55,
    textAlign: 'center'
  }
})

export default function TextBilingualHeading(props) {
  return (
    <Fragment>
      <Title style={style.title}>{props.heading.title}</Title>
      <Subheading style={style.author}>{props.heading.author}</Subheading>
      <Caption style={style.source}>{props.heading.source}</Caption>
    </Fragment>
  )
}

TextBilingualHeading.propTypes = {
  heading: PropTypes.shape({
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    source: PropTypes.string.isRequired
  })
}
