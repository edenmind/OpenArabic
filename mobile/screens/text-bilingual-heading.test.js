import { describe, it, jest } from '@jest/globals'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import React from 'react'
import TextBilingualHeading from './text-bilingual-heading.js'

jest.useFakeTimers()
describe('<Heading />', () => {
  it('renders without crashing', () => {
    const propsMock = {
      title: 'title',
      author: 'author',
      source: 'source',
      readTime: 'readTime'
    }

    // @ts-ignore
    const renderer = new ShallowRenderer()

    renderer.render(<TextBilingualHeading heading={propsMock} />)
  })
})
