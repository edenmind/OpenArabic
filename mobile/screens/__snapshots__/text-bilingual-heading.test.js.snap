// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBilingualHeading /> renders correctly 1`] = `
[
  <Image
    source={
      {
        "uri": "image",
      }
    }
    style={
      {
        "height": 250,
        "width": "100%",
      }
    }
  />,
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 28,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 36,
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "philosopher",
            "paddingHorizontal": 25,
            "paddingTop": 25,
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    title
  </Text>,
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          {
            "color": "rgba(73, 69, 79, 1)",
            "marginLeft": 35,
            "marginRight": 35,
            "paddingTop": 15,
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    author
  </Text>,
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "500",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          {
            "color": "rgba(73, 69, 79, 1)",
            "marginLeft": 35,
            "marginRight": 35,
            "paddingTop": 15,
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    source
  </Text>,
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 16,
          },
          {
            "color": "rgba(121, 116, 126, 1)",
            "paddingBottom": 10,
            "paddingTop": 20,
            "textAlign": "center",
          },
        ],
      ]
    }
  >
    5 min read · undefined views · undefined
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "rgba(202, 196, 208, 1)",
          "height": 0.5,
        },
        undefined,
        false,
        false,
        {
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 25,
        },
      ]
    }
  />,
]
`;
